src/number_theory/legendre_symbol/quadratic_char.lean

```diff
@@ -18,11 +19,22 @@ quadratic character
 -/
 
 /-!
-### Some general results on finite fields
+### Some general results, mostly on finite fields
+
+We collect some results here that are not specific to quadratic characters
+but are needed below. They will be moved to appropriate places eventually.
 -/
 
 section general
 
+/-- An odd natural number has residue `1` or `3` mod `4`-/
+lemma odd_mod_four {n : ℕ} (hn : n % 2 = 1) : n % 4 = 1 ∨ n % 4 = 3 :=
```

Comment:
This can be turned into an `iff` (or maybe we want both versions). As `iff` statement, it can be used in `rw`, which I guess will be useful.

Reply:
Makes sense; I'll change to `odd_mod_four_iff`. Should this go into the `nat` namespace?

Reply:
I've moved it into the `nat` namespace (where the partial converses also reside).